### ISSUE TITLE
plugin syntax-jsx cannot parse arrow function to JSXIdentifier inside function

### ISSUE THREAD

**Issue by huozhi** [babel/minify#948]
## Bug Report

**Current Behavior**
use babel  with `@babel/plugin-syntax-jsx` plugin to parse the following code snippet will lead to errors.

```
TypeError: Property name of JSXOpeningElement expected node to be of a type ["JSXIdentifier","JSXMemberExpression"] but instead got "ArrowFunctionExpression"
```

**Input Code**

snippet.js

```js
it("should support nested Profilers", () => {
  const FunctionComponent = ({ label }) => <div>{label}</div>;
  return <FunctionComponent label="outer function component" />
});
```

but if you remove the top `it` function it will be parsed.

**Expected behavior/code**
parse successfully without errors

**Babel Configuration (.babelrc, package.json, cli command)**

```js
./node_modules/.bin/babel --plugins @babel/plugin-syntax-jsx snippet.js
```

**Environment**
- Babel version(s): v7.3.4
- Node/npm version: [Node 11/npm 6]
- OS: [OSX 10.13.6]
- How you are using Babel: cli

**Possible Solution**

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by danez** [comment]
Yes it seems that dead-code-elimination does not handle jsx correctly. I was able to reproduce. in the repl by adding `@babel/plugin-syntax-jsx` and `babel-plugin-minify-dead-code-elimination`

Also transfered the issue.
